### ISSUE TITLE
Issues running a plan with asm module if cluster is not created yet

### ISSUE THREAD

**Issue by DP19** [terraform-google-modules/terraform-google-kubernetes-engine#1364]
### TL;DR

Running a plan currently causes the same error seen in https://github.com/terraform-google-modules/terraform-google-gcloud/issues/82

Even with the changes added in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1354

If we remove the `depends_on` from the asm module call leaving only the `module_depends_on` we get an error about the google_container_cluster.asm.id being null

```
  on .terraform/modules/gke_cluster_jobs.kubernetes_engine_asm/modules/asm/hub.tf line 24, in resource "google_gke_hub_membership" "membership":
  24:       resource_link = "//container.googleapis.com/${data.google_container_cluster.asm.id}"
    |----------------
    | data.google_container_cluster.asm.id is null
```

can we also add a `depends_on` block to this data resource? Adding this in manually to my locally copied version of the asm module produces a plan correctly 

### Terraform Configuration

```hcl
module "kubernetes_engine_asm" {
  module_depends_on         = [module.gke_cluster]
  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  version                   = "22.1.0"
  project_id                = var.project_name
  channel                   = lower(var.release_channel)
  cluster_location          = var.cluster_location
  cluster_name              = var.cluster_name
  enable_cni                = true
  enable_fleet_registration = true
  enable_mesh_feature       = true
}
```

```
module "gke_cluster" {
  source                          = "../../../modules/gcp/gke/cluster-v2"
...
}
```
```


### Terraform Version

```sh
Terraform v0.14.11
+ provider registry.terraform.io/hashicorp/external v2.2.2
+ provider registry.terraform.io/hashicorp/google v4.25.0
+ provider registry.terraform.io/hashicorp/google-beta v4.25.0
+ provider registry.terraform.io/hashicorp/helm v2.6.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.12.1
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.3.2
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

**Comment by DP19** [comment]
@apeabody I know there's a release pr up that will have the changes in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1354 live; hoping to have this fix along with it 🤞🏻

**Comment by apeabody** [comment]
Thanks @DP19
